### PR TITLE
CA-233306: Delete the VDI when invoking Pool_update.pool_clean

### PR DESCRIPTION
**Pull request by huizh**
Db.VDI.destroy in Pool_update.pool_clean just removes the Xapi database record.
It should invoke Client.VDI.destroy instead and remove both the VDI and the database record.

Signed-off-by: Hui Zhang <hui.zhang@citrix.com>

### CODE REVIEW

**Comment by lindig** [comment]
This looks good. I assume, `Client.VDI.destroy` also removes the Xapi database record. If so, I'd rephrase the commit message to: 

> Db.VDI.destroy in `pool_clean` just removes the Xapi database record.
> It should invoke Client.VDI.destroy instead and remove both the VDI and the database record.

**Comment by huizh** [comment]
Thanks @lindig. Commit message is updated.
